app/javascript/packs/BraintreeForm.js

```diff
@@ -0,0 +1,35 @@
+import {
+  BraintreeFormWrapper,
+  createBraintreeClient
+} from 'PaymentGateways'
+import client from 'braintree-web/client'
+import { safeFromJsonString } from 'utilities/json-utils'
+
+const CONTAINER_ID = 'braintree-form-wrapper'
+
+document.addEventListener('DOMContentLoaded', async () => {
+  const container = document.getElementById(CONTAINER_ID)
+  if (!container) {
+    return
+  }
+
+  const { clientToken, billingAddress, threeDSecureEnabled, formActionPath, countriesList, selectedCountryCode } = container.dataset
+  const billingAddressParsed = safeFromJsonString(billingAddress)
+  for (let key in billingAddressParsed) {
+    if (!billingAddressParsed[key]) {
```

Comment:
Could we make this more explicit? `!something` could mean `something === ''`. What's being checked here exactly?

Reply:
Some of this properties may be `null` (from rails), I want all values to be strings, to avoid checking if they are null or strings